src/pke/lib/scheme/bfvrns/bfvrns-leveledshe.cpp

```diff
@@ -762,13 +789,17 @@ std::shared_ptr<std::vector<DCRTPoly>> LeveledSHEBFVRNS::EvalFastRotationPrecomp
     const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersBFVRNS>(ciphertext->GetCryptoParameters());
     auto algo               = ciphertext->GetCryptoContext()->GetScheme();
 
-    if (cryptoParams->GetMultiplicationTechnique() != HPSPOVERQLEVELED) {
+    size_t sizeQ             = ciphertext->GetElements()[0].GetNumOfElements();
+    const auto elementParams = cryptoParams->GetElementParams();
+    // Maximum number of RNS limbs in the crypto context
+    size_t sizeQM = elementParams->GetParams().size();
+
+    if (!((cryptoParams->GetMultiplicationTechnique() == HPSPOVERQLEVELED) && (sizeQ == sizeQM))) {
         return algo->EvalKeySwitchPrecomputeCore(ciphertext->GetElements()[1], ciphertext->GetCryptoParameters());
     }
 
     DCRTPoly c1     = ciphertext->GetElements()[1];
```

Comment:
Is the part for when (cryptoParams->GetMultiplicationTechnique() == HPSPOVERQLEVELED) && (sizeQ == sizeQM) still required?

Reply:
Still needed, but restructured the code to look simpler